# NMPC横向控制器参数配置
nmpc_lateral_controller:
  # 状态与控制维度配置
  nx: 6                 # 状态维度 [x, y, theta, gamma, ltr, dltr]
  nu: 1                 # 控制量维度 [delta]
  
  # 预测与控制步长配置
  prediction_step: 20   # NMPC预测步长
  sparse_control_step: 4 # 稀疏控制量步数
  
  # 车辆动力学参数
  a1: -1.5               # 示例默认值，需根据实际车型调整
  a2: -1.5               # 示例默认值
  b1: 0.001               # 关键：加载 b1_，设默认值避免随机
  
  sampling_time: 0.1              # 采样时间 (s)
  tractor_wheelbase: 2.8          # 牵引车轴距 (m)
  trailer_wheelbase: 11.5         # 挂车轴距 (m)
  gravity: 9.806                  # 重力加速度 (m/s²)
  max_acceleration: 0.5           # 预设最大加速度 (m/s²)
  
  # 参考路径处理参数
  num_waypoints: 30               # 目标参考路点数量
  
  # 控制量边界约束
  steer_limit: 0.4                # 最大转向角 (rad)

  # 状态量边界约束
  ltr_limit: 0.6                  # 最大侧向载荷转移率
  
  # 代价函数权重
  weight_position: 55.0            # 位置跟踪权重
  weight_heading: 13.0             # 航向跟踪权重
  weight_front_steer: 500.0       # 前轴转向角平滑权重
  weight_terminal_position: 10.0  # 终端位置权重
  weight_terminal_heading: 5.0    # 终端航向权重
  weight_ltr_constraint: 10000.0  # 侧向载荷转移率约束松弛变量权重
    



pid_controller:
  kp: 0.6
  ki: 0.05
  kd: 0.1
  max_throttle: 0.8
  max_brake: 0.8
  integral_limit: 2.0
  speed_tolerance: 0.2
  min_target_speed: 0.5
  predict_time_horizon: 0.5  # 预测时间 Horizon（s），默认0.5（用于将未来速度给到轨迹追踪，辅助克服控制时延）